rfdetr/datasets/yolo.py

```diff
@@ -0,0 +1,360 @@
+
+import torch
+import torch.utils.data
+
+from rfdetr.datasets.coco import make_coco_transforms, make_coco_transforms_square_div_64
+from rfdetr.util.files import load_txt_lines
+
+REQUIRED_YOLO_YAML_FILE = "data.yaml"
+REQUIRED_SPLIT_DIRS = ["train", "valid"]
+REQUIRED_DATA_SUBDIRS = ["images", "labels"]
+
+
+def is_valid_yolo_dataset(dataset_dir: str) -> bool:
+    """
+    Checks if the specified dataset directory is in yolo format.
+
+    We accept a dataset to be in yolo format if the following conditions are met:
+    - The dataset_dir contains a data.yaml file
+    - The dataset_dir contains "train" and "valid" subdirectories, each containing "images" and "labels" subdirectories
+    - The "test" subdirectory is optional
+
+    Returns a boolean indicating whether the dataset is in correct yolo format.
+    """
+    contains_required_data_yaml = os.path.exists(os.path.join(dataset_dir, REQUIRED_YOLO_YAML_FILE))
+    contains_required_split_dirs = all(
+        os.path.exists(os.path.join(dataset_dir, split_dir)) for split_dir in REQUIRED_SPLIT_DIRS
+    )
+    contains_required_data_subdirs = all(
+        os.path.exists(os.path.join(dataset_dir, split_dir, data_subdir))
+        for split_dir in REQUIRED_SPLIT_DIRS
+        for data_subdir in REQUIRED_DATA_SUBDIRS
+    )
+    return contains_required_data_yaml and contains_required_split_dirs and contains_required_data_subdirs
+
+
+def build_yolo(image_set, args, resolution):
+    """Build YOLO dataset"""
+    root = Path(args.dataset_dir)
+    print(image_set)
+    PATHS = {
+        "train": (root / "train" / "images", root / "train" / "labels"),
+        "val": (root / "valid" / "images", root / "valid" / "labels"),
+        "test": (root / "test" / "images", root / "test" / "labels"),
+    }
+    
+    img_folder, labels_folder = PATHS[image_set.split("_")[0]]
+    data_yaml_path = root / "data.yaml"
+    
+    try:
+        square_resize_div_64 = args.square_resize_div_64
+    except:
+        square_resize_div_64 = False
+    
+    if square_resize_div_64:
+        dataset = YoloDetection(
+            img_folder, 
+            labels_folder,
+            data_yaml_path,
+            transforms=make_coco_transforms_square_div_64(
+                image_set, 
+                resolution, 
+                multi_scale=args.multi_scale, 
+                expanded_scales=args.expanded_scales
+            )
+        )
+    else:
+        dataset = YoloDetection(
+            img_folder, 
+            labels_folder,
+            data_yaml_path,
+            transforms=make_coco_transforms(
+                image_set, 
+                resolution, 
+                multi_scale=args.multi_scale, 
+                expanded_scales=args.expanded_scales
+            )
+        )
+    
+    return dataset 
+
+
+class YoloDetection(torch.utils.data.Dataset):
+    """Dataset for YOLO format annotations"""
+    def __init__(self, img_folder, labels_folder, data_yaml_path, transforms=None):
```

Comment:
Let's rename `img_folder` and `labels_folder` to `images_directory_path` and `annotations_directory_path`.